clients/src/main/java/org/apache/kafka/common/protocol/Errors.java

```diff
@@ -305,7 +306,9 @@
             PreferredLeaderNotAvailableException::new),
     GROUP_MAX_SIZE_REACHED(81, "The consumer group has reached its max size.", GroupMaxSizeReachedException::new),
     FENCED_INSTANCE_ID(82, "The coordinator reports a more recent member.id associated with the consumer's group.instance.id.",
-            FencedInstanceIdException::new);
+            FencedInstanceIdException::new),
+    PARTITION_LEADER_NOT_AVAILABLE(83, "A topic partition leader was not available",
```

Comment:
Yeah. I though about using that error code. The felt that the string message reported by the protocol is miss leading. Looking at the code, this error is mainly used by the broker to report errors during a fetch. Here we are trying to report that the controller couldn't elect a leader because all of the replicas were unavailable.

I am okay reusing the error and/or making the string message a bit more generic. Let me know what you think.

Reply:
Yeah, the use case is a bit different. Perhaps we just need a different name. Perhaps `NO_ELIGIBLE_LEADERS_AVAILABLE` or something like that?